src/mixins/document-mixin.ts

```diff
@@ -38,4 +49,24 @@ export default class DocumentMixin extends Vue {
         return error.response
       })
   }
+
+  /**
+   * Checks whether pdf file is using specified page size by checking width and height of 1st page of pdf file
+   * @param file pdf file to be checked
+   * @param pageSize enum value used to represent page size to check for
+   * @return Promise<boolean> whether pdf file is expected page size
+   */
+  async isPageSize (file, pageSize: PdfPageSize): Promise<boolean> {
+    const pageSizeInfo = this.pageSizeInfo[pageSize]
+    const pdfBufferData = await file.arrayBuffer()
+    const pdfData = new Uint8Array(pdfBufferData) // put it in a Uint8Array
+    const pdf = await pdfjsLib.getDocument({ data: pdfData })
+    const p1 = await pdf.getPage(1)
+    const [x, y, w, h] = p1._pageInfo.view
+    const width = w - x
+    const height = h - y
+    const isvalidPageSize = (width / pageSizeInfo.pointsPerInch === pageSizeInfo.width) &&
+      (height / pageSizeInfo.pointsPerInch === pageSizeInfo.height)
+    return isvalidPageSize
```

Comment:
Nice code!

Are you sure those 3 calls are async?

Reply:
pretty sure they are all async but will double check

Reply:
confirmed all 3 are async